### DISCUSSION TITLE
Passing hex data rather than ASCII

### DISCUSSION THREAD

**Discussion by denisdowney** [armadsen/ORSSerialPort#177] [Q&A] [answered]
I'm interfacing with a device that sends a stream of hex data (starts with 0x02, ends with 0x03).  As far as I can tell, ORSSerialPort is configured for ASCII by default.  Is there a way to configure for binary/hex ?

**Comment by armadsen** [comment, marked as answer]
Hi Denis,

ORSSerialPort is not configured for ASCII by default. It deals with raw binary data, as represented by instances of `NSData` (`Data` in Swift). There are some convenience APIs in a couple places to make sending and receiving strings easier, but they're optional to use and all have `(NS)Data`-based counterparts. What exactly are you running into trouble with?

**Comment by denisdowney** [comment]
Thanks Andrew,

I was focused on the NSString references, rather than NSData.  Should have
dug into the code.

Thanks for your explanation.

Denis
